scripts/storm_control.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+
+############################################
+#
+# script to test storm_control functionality
+#
+############################################
+
+import argparse
+import sys
+import os
+
+# mock the redis for unit test purposes #
+try:
+    if os.environ["UTILITIES_UNIT_TESTING"] == "2":
+        modules_path = os.path.join(os.path.dirname(__file__), "..")
+        test_path = os.path.join(modules_path, "tests")
+        sys.path.insert(0, modules_path)
+        sys.path.insert(0, test_path)
+        import mock_tables.dbconnector
+
+except KeyError:
+    pass
+
+from natsort import natsorted
+from tabulate import tabulate
+from swsscommon.swsscommon import SonicV2Connector, ConfigDBConnector
+from utilities_common.general import load_db_config
+
+STORM_TABLE_NAME = "PORT_STORM_CONTROL"
+
+class storm_control(object):
+    def __init__(self):
+        self.config_db = ConfigDBConnector()
+        self.config_db.connect()
+        self.db = SonicV2Connector(use_unix_socket_path=False)
+        self.db.connect(self.db.CONFIG_DB)
+    def show_storm_config(self, port):
+        header = ['Interface Name', 'Storm Type', 'Rate (kbps)']
+        storm_type_list = ['broadcast','unknown-unicast','unknown-multicast']
+        body = []
+        configs = self.db.get_table(STORM_TABLE_NAME)
+        if not configs:
+            return
+        storm_configs = natsorted(configs)
+        if port is not None:
+            for storm_type in storm_type_list:
+                storm_key = port + '|' + storm_type
+                data = self.db.get_entry(STORM_TABLE_NAME, storm_key)
+                if data:
+                    kbps = data['kbps']
+                    body.append([port, storm_type, kbps])
+        else:
+            for storm_key in storm_configs:
+                interface_name = storm_key[0]
+                storm_type = storm_key[1]
+                data = self.db.get_entry(STORM_TABLE_NAME, storm_key)
+                if data:
+                    kbps = data['kbps']
+                    body.append([interface_name, storm_type, kbps])
+        print(tabulate(body,header,tablefmt="grid")) 
+
+    def validate_interface(self, port):
+        if not (port.startswith("Eth")):
+            return False
+        return True
+
+    def validate_kbps(self, kbps):
+        return True
+
+    def add_storm_config(self, port, storm_type, kbps):
```

Comment:
How do you handle thec ase where if the HWSKU does not support this rate limiting on the interface?
You should add query capability on the port attribute to ensure that the attribute is supported instead of blindly accpet the configuration and failed at Orchagent layer.
I do h=not see this being considered at the HLD either.
Please add this so that the configuration will not be allowed if the HWSKU being configured does not support such operation.

Reply:
Added code to store capability in STATE_DB and check before proceeding with CONFIG_DB update.